File: fincapp-backend/src/FincApp.API/DTOs/SyncAnimalDto.cs
using System;
using System.Text.Json.Serialization;
using FincApp.Core;

namespace FincApp.API.DTOs;

public class SyncAnimalDto
{
    [JsonPropertyName("id")]
    public Guid Id { get; set; }

    [JsonPropertyName("type")]
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public ProductionType Type { get; set; }

    [JsonPropertyName("identification_tag")]
    public string IdentificationTag { get; set; } = string.Empty;

    [JsonPropertyName("birth_date")]
    public DateTime? BirthDate { get; set; }

    [JsonPropertyName("status")]
    public string Status { get; set; } = "healthy";
}

File: fincapp-frontend/mobile-app/app/src/main/java/com/irwi/fincapp/network/dto/SyncAnimalDto.java
package com.irwi.fincapp.network.dto;

import com.google.gson.annotations.SerializedName;

public class SyncAnimalDto {
    public String id;
    public String type;
    @SerializedName("identification_tag") public String identificationTag;
    @SerializedName("birth_date") public String birthDate;
    public String status;
}
